cli/src/lib/paths.js

```diff
@@ -7,20 +7,21 @@ const shellSource = path.dirname(
 )
 const shellAppDirname = 'src/D2App'
 
-const findYarnLock = (base) => {
-    if (base === '/') {
+const findYarnLock = (base, directoryRoot) => {
+    if (base === directoryRoot) {
         return null
     }
-
     const yarnLock = path.join(base, './yarn.lock')
     if (fs.existsSync(yarnLock)) {
         return yarnLock
     }
-    return findYarnLock(path.dirname(base))
+    return findYarnLock(path.dirname(base), directoryRoot)
 }
 
 module.exports = (cwd = process.cwd()) => {
     const base = path.resolve(cwd)
+    // Get the root path or drive based on the operating system
+    const rootPathByOS = path.parse(cwd).root;
```

Comment:
This is no longer dependent on the operating system, it should work for all operating systems, so we should remove this comment and change the name to simply `rootDirectory`

```suggestion
    const rootDirectory = path.parse(cwd).root;
```